staging/src/k8s.io/apiserver/pkg/storage/value/helpers.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package value
+
+import (
+	"fmt"
+)
+
+// KeyConfig contains the name of the key, and the file in which it is stored
+type KeyConfig struct {
+	Name   string
+	Secret string
+}
+
+// GetKeyDataFromConfig parses a valid interface to a KeyConfig object
+func GetKeyDataFromConfig(config interface{}) (KeyConfig, error) {
```

Comment:
This is too generic - it's better to have a package that takes a concrete config api (i.e. Kube API structs) type and then knows how to turn that into a concrete AES PrefixTransformer, and then have  a fairly top level package in apiserver be consumable (pkg/storage/encryption/options.go) to pass into the StorageConfig.

You can define your config api types in a place  where config apis would generally belong - however, I know we are trying not to add more things to kubernetes/pkg/apis/componentconfig.  @deads2k where would api server specific config types go today?